jupiter-tests/src/test/java/org/junit/jupiter/engine/extension/DefaultTimeZoneTests.java

```diff
@@ -0,0 +1,378 @@
+		}
+
+	}
+
+	@Nested
+	@DefaultTimeZone("GMT")
+	@DisplayName("when explicitly set to GMT on the class level")
+	class ExplicitGmtClassLevelTestCases {
+
+		@Test
+		@DisplayName("does not throw and sets to GMT ")
+		void explicitGmt() {
+			assertThat(TimeZone.getDefault()).isEqualTo(TimeZone.getTimeZone("GMT"));
+		}
+
+	}
+
+	@Nested
+	@DefaultTimeZone("GMT-8:00")
+	@DisplayName("with nested classes")
+	class NestedTests {
+
+		@Nested
+		@DisplayName("without DefaultTimeZone annotation")
+		class NestedClass {
+
+			@Test
+			@ReadsDefaultTimeZone
+			@DisplayName("DefaultTimeZone should be set from enclosed class when it is not provided in nested")
+			public void shouldSetTimeZoneFromEnclosedClass() {
+				assertThat(TimeZone.getDefault()).isEqualTo(TimeZone.getTimeZone("GMT-8:00"));
+			}
+
+		}
+
+		@Nested
+		@DefaultTimeZone("GMT-12:00")
+		@DisplayName("with DefaultTimeZone annotation")
+		class AnnotatedNestedClass {
+
+			@Test
+			@ReadsDefaultTimeZone
+			@DisplayName("DefaultTimeZone should be set from nested class when it is provided")
+			public void shouldSetTimeZoneFromNestedClass() {
+				assertThat(TimeZone.getDefault()).isEqualTo(TimeZone.getTimeZone("GMT-12:00"));
+			}
+
+			@Test
+			@DefaultTimeZone("GMT-6:00")
+			@DisplayName("DefaultTimeZone should be set from method when it is provided")
+			public void shouldSetTimeZoneFromMethodOfNestedClass() {
+				assertThat(TimeZone.getDefault()).isEqualTo(TimeZone.getTimeZone("GMT-6:00"));
+			}
+
+		}
+
+	}
+
+	@Nested
+	@DefaultTimeZone("GMT-12:00")
+	@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+	class ResettingDefaultTimeZoneTests {
+
+		@Nested
+		@DefaultTimeZone("GMT-3:00")
+		@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+		class ResettingDefaultTimeZoneNestedTests {
+
+			@Test
+			@DefaultTimeZone("GMT+6:00")
+			void setForTestMethod() {
+				// only here to set the time zone, so another test can verify whether it was reset;
+				// still, better to assert the value was actually set
+				assertThat(TimeZone.getDefault()).isEqualTo(TimeZone.getTimeZone("GMT+6:00"));
+			}
+
+			@AfterAll
+			@ReadsDefaultTimeZone
+			void resetAfterTestMethodExecution() {
+				assertThat(TimeZone.getDefault()).isEqualTo(TEST_DEFAULT_TIMEZONE);
+			}
+
+		}
+
+		@AfterAll
+		@ReadsDefaultTimeZone
+		void resetAfterTestMethodExecution() {
+			assertThat(TimeZone.getDefault()).isEqualTo(TEST_DEFAULT_TIMEZONE);
+		}
+
+	}
+
+	@Nested
+	@DisplayName("when misconfigured")
+	class ConfigurationTests {
+
+		@Test
+		@ReadsDefaultTimeZone
+		@DisplayName("on method level, throws exception")
+		void throwsWhenConfigurationIsBad() {
+			ExecutionResults results = executeTestMethod(BadMethodLevelConfigurationTestCases.class,
+				"badConfiguration");
+
+			assertThat(results).hasSingleFailedTest().withExceptionInstanceOf(
+				ExtensionConfigurationException.class).hasMessageNotContaining(
+					"should never execute").hasMessageContaining("@DefaultTimeZone not configured correctly.");
```

Comment:
Please remove the classes in `org.junit.jupiter.testkit.assertion.**` and use "regular" `EngineTestKit` assertions:

```suggestion
			results.testEvents().assertThatEvents() //
					.haveExactly(1, finishedWithFailure(instanceOf(ExtensionConfigurationException.class),
						message(it -> it.contains("@DefaultTimeZone not configured correctly."))));
```